packages/fxa-settings/src/components/BrandMessaging/index.tsx

```diff
@@ -0,0 +1,144 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React, { useEffect, useState } from 'react';
+import { useConfig } from '../../models';
+import { logViewEventOnce } from '../../lib/metrics';
+import { navigate } from '@reach/router';
+import { ReactComponent as IconM } from './icon_m.svg';
+import { ReactComponent as IconClose } from './icon_close.svg';
+import { FtlMsg } from 'fxa-react/lib/utils';
+
+export const bannerClosedLocalStorageKey =
+  '__fxa_storage.fxa_disable_brand_banner';
+
+const BrandMessaging = ({
+  mode,
+  viewName,
+}: {
+  mode?: string;
+  viewName: string;
+}) => {
+  const { brandMessagingMode } = useConfig();
+
+  if (mode === undefined) {
+    mode = brandMessagingMode;
+  }
+
+  const [bannerClosed, setBannerClosed] = useState<string | null>(
+    localStorage.getItem(`${bannerClosedLocalStorageKey}_${mode}`)
+  );
+
+  const bannerVisible =
+    !bannerClosed && (mode === 'prelaunch' || mode === 'postlaunch');
+
+  useEffect(() => {
+    if (bannerVisible) {
+      // hack to ensure banner doesn't overlap, since the only reasonable to way to do this was
+      // with fixed layout.
+      setTimeout(() => {
+        document.body.classList.add('brand-messaging');
+      }, 0);
+    }
+  });
+
+  // Short circuit if mode is unknown or user has previously closed banner
+  if (!bannerVisible) {
+    return <></>;
+  }
+
+  logViewEventOnce(`flow.${viewName}`, `brand-messaging-${mode}-view`);
+
+  function onClickLearnMore() {
+    logViewEventOnce(`flow.${viewName}`, `brand-messaging-${mode}-learn-more`);
+    navigate(
+      'https://support.mozilla.org/kb/firefox-accounts-renamed-mozilla-accounts'
+    );
+  }
+
+  function onClickCloseBanner() {
+    logViewEventOnce(
+      `flow.${viewName}`,
+      `brand-messaging-${mode}-banner-close`
+    );
```

Comment:
I don't see any mapping to "amplitude" events for the flow events in this PR.  I'm not sure if that's needed/wanted.

Reply:
I was a little uncertain about this... I might side bar with you on this. I assumed the flow event would be enough... but maybe not? The ticket didn't have much info about how to emit the event or what to call it. Do you have any advice here?

Reply:
In that case let's add them.  It involves adding mappings to `EVENTS` in https://github.com/mozilla/fxa/blob/main/packages/fxa-content-server/server/lib/amplitude.js  As for names since they weren't specified let's just use whatever make sense.

Reply:
Unfortunately because the metrics code inserts the view name into those events, e.g. `flow.enter-email.brand-messaging-prelaunch-banner-close`, you gonna need to use regex.  That means adding the mapping to `FUZZY_EVENTS` instead of `EVENTS`.  Sorry about the misinformation earlier.  :(

Reply:
No worries, thanks for looking at it again. I've updated `FUZZY_EVENTS`, and tweaked tests a bit. (In hindsight this makes sense... of course view name would need to be added in somehow.)